plugin/trino-clickhouse/src/test/java/io/trino/plugin/clickhouse/IsNullPushdownDataTypeTest.java

```diff
@@ -0,0 +1,209 @@
+        }
+        catch (AssertionError e) {
+            // if failed - identify exact column which caused the failure
+            for (int column = SPECIAL_COLUMNS; column < testCases.size(); column++) {
+                String queryWithSingleColumnPredicate = "SELECT " + firstColumnName + " FROM " + temporaryRelation.getName() + " WHERE " + getPredicate(column, scenario) + withConnectorExpression;
+                assertPushdown(expectPushdown,
+                        assertResult(scenario.equals(IS_NULL) ? Optional.of(firstCase.expectedLiteral()) : Optional.empty(),
+                                assertThat(queryAssertions.query(session, queryWithSingleColumnPredicate))));
+            }
+            throw new IllegalStateException("Single column assertion should fail for at least one column, if query of all column failed", e);
+        }
+    }
+
+    private String getPredicate(int column, Scenario scenario)
+    {
+        checkArgument(column >= SPECIAL_COLUMNS, "Special columns should not be a part of a predicate, as they are helpers and do not participate in the assertions");
+        String columnName = "col_" + column;
+        checkArgument("NULL".equalsIgnoreCase(testCases.get(column).inputLiteral()));
+        return scenario == IS_NULL
+                ? columnName + " IS NULL"
+                : columnName + " IS NOT NULL";
+    }
+
+    private static QueryAssert assertResult(Optional<String> value, QueryAssert assertion)
+    {
+        return value.isPresent()
+                ? assertion.matches("VALUES %s".formatted(value.get()))
+                : assertion.returnsEmptyResult();
+    }
+
+    private static QueryAssert assertPushdown(boolean expectPushdown, QueryAssert assertion)
+    {
+        return expectPushdown
+                ? assertion.isFullyPushedDown()
+                : assertion.isNotFullyPushedDown(FilterNode.class);
+    }
+
+    private record TestCase(
+            String declaredType,
+            String inputLiteral,
+            Type expectedType,
+            String expectedLiteral)
+            implements ColumnSetup
+    {
+        private TestCase
+        {
+            requireNonNull(declaredType, "declaredType is null");
+            requireNonNull(expectedType, "expectedType is null");
+            requireNonNull(inputLiteral, "inputLiteral is null");
+            requireNonNull(expectedLiteral, "expectedLiteral is null");
+        }
+
+        @Override
+        public Optional<String> getDeclaredType()
+        {
+            return Optional.of(declaredType);
+        }
+
+        @Override
+        public String getInputLiteral()
+        {
+            return inputLiteral;
+        }
+    }
+
+    enum Scenario
+    {
+        IS_NULL,
+        IS_NOT_NULL,
+    }
+
+    public enum PushdownImplementation
+    {
+        TUPLE_DOMAIN_ONLY,
+        CONNECTOR_EXPRESSION_ONLY,
+        ALWAYS,
+    }
```

Comment:
Do we need three states here ? Ideally the pushdown should happen for both tuple domain and connector expression - For now we have a small gaps where tuple domain is not supported - but it could removed in future - so it would `ALWAYS` - always - For now we can we have a boolean which would say `onlyConnectorExpression` which would be removed in the future